src/components/SearchResults.js

```diff
@@ -91,6 +97,94 @@ export default class SearchResults extends React.Component<SearchResultsDefaultP
 
   static displayName = 'SearchResults';
 
+  constructor(props: SearchResultsProps) {
+    super(props);
+    this.state = {
+      activeTableTabKey: 'All',
+    };
+
+    (this: any).wrapResultsInTabs = this.wrapResultsInTabs.bind(this);
+    (this: any).handleTabChange = this.handleTabChange.bind(this);
+  }
+
+  wrapResultsInTabs(results: Array<any>, response: QueryResponse) {
+    const facets = response.facets;
+    let tables = [];
+    let tableFacet;
+    facets.forEach((f: SearchFacet) => {
+      if (f.name === this.props.tabsField) {
+        tableFacet = f;
+      }
+    });
+
+    let { activeTableTabKey } = this.state;
+    if (!this.context.searcher.state.filters || this.context.searcher.state.filters.length <= 0) {
+      activeTableTabKey = 'All';
+    } else {
+      this.context.searcher.state.filters.forEach((f: string) => {
+        if (f.indexOf('table') > -1) {
+          activeTableTabKey = f;
+        }
+      });
+    }
+    if (this.props.tabList.length < 1 && tableFacet && tableFacet.buckets) {
+      tables = tableFacet.buckets.map((bucket: SearchFacetBucket) => {
+        const innerContents = activeTableTabKey === bucket.filter ? results : '';
+        return (
+          <Tab eventKey={bucket.filter} title={`${bucket.value} (${bucket.count})`}>
+            {innerContents}
+          </Tab>
+        );
+      });
+    } else if (this.props.tabList.length > 1) {
+      tables = this.props.tabList.map((tableName: string) => {
+        let isDisabled = true;
+        let count = 0;
+        let innerContents;
+        let eventKey = tableName;
+        if (tableFacet && tableFacet.buckets) {
+          tableFacet.buckets.forEach((facet: SearchFacetBucket) => {
+            if (facet.value === tableName) {
+              isDisabled = false;
+              count = facet.count;
+              eventKey = facet.filter;
+              if (activeTableTabKey === facet.filter) {
+                innerContents = results;
+              }
+            }
+          });
+        }
+        return (
+          <Tab eventKey={eventKey} title={`${tableName} (${count})`} disabled={isDisabled}>
+            {innerContents}
+          </Tab>
+        );
+      });
+    }
+    if (tables && tables.length > 0) {
+      const innerContents = activeTableTabKey === 'All' ? results : '';
+      tables.unshift(
+        <Tab eventKey="All" title="All">
```

Comment:
I might use a different (more complex/unusual) value for the key here, on the off chance that one of the bucket's is called "All"